sisl/viz/plotly/plots/grid.py

```diff
@@ -1062,7 +1062,7 @@ def scan(self, along=None, start=None, stop=None, steps=None, breakpoints=None,
                 step = (along_range[1] - along_range[0])/steps
             elif isinstance(steps, float):
                 step = steps
-                steps = (along_range[1] - along_range[0])/step
+                steps = int((along_range[1] - along_range[0]) / step)
```

Comment:
I've changed this. However apparently `float // int` returns a floored `float`, not an `int`. I.e. `3.9 // 2` == `1.0`. So I had to still convert it to an `int` afterwards :sweat_smile:

Reply:
Also, I have separated the argument into two arguments: `step` and `steps`. Is that ok?

Reply:
Hmm, that might also be confusion.

If you want both, then `step` (from `np.arange`) and `num` (from `np.linspace`) might be better since they are more clear in intent. You should then assert that not both are passed ;)

Great with the `int` comment, I had forgotten that ;)
Well, I like the `//` in any case. It clarifies intent.

Reply:
> You should then assert that not both are passed ;)

Isn't it enough to state in the docs that one will be ignored if the other is passed?

Reply:
Ok, to me `num` is more confusing than `steps`, but if thats how `np.linspace` calls it...

Reply:
> > You should then assert that not both are passed ;)
> 
> Isn't it enough to state in the docs that one will be ignored if the other is passed?

No, always raise `ValueError` if user does not obey this.
While it may seem obvious what would happen it isn't and that can lead to wrong behaviour.

> Ok, to me `num` is more confusing than `steps`, but if thats how `np.linspace` calls it...

Yeah, well it was just to select a "standard" ;) And numpy is pretty good here ;)

Reply:
> No, always raise ValueError if user does not obey this.

now? :)